src/pages/api/token-details.ts

```diff
@@ -0,0 +1,32 @@
+import config from '@/config'
+import type { NextApiRequest, NextApiResponse } from 'next'
+
+type ResponseData = {
+  status: boolean
+  message: string
+  response?: any
+}
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<ResponseData>,
+) {
+  const { tokenAddress, chain } = req.query
```

Comment:
this is not a token address, you are looking for the tokens inside of a wallet. walletAddress might be less misleading